### DISCUSSION TITLE
Improve efficiency of sidewalk quest

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#2684] [Ideas]
It's finally warm enough for me to get back into mapping!

Right now the sidewalk quest is 5 taps to answer: tap left, select y/n, tap right, select y/n, OK.

Instead, it could be 2 taps (road drawn at a slant to indicate that the icons should rotate with the map):

![image](https://user-images.githubusercontent.com/6563664/112665405-ed9ec580-8e52-11eb-8d2c-7de286926821.png)

If that wouldn't work on smaller devices, it could be a 2x2 grid instead.

I also wonder if a similar thing could be applied to the cycleway quest and any other quests which apply to both sides (#2560, #2193). For example, making it like the lanes quest, a two-part question where the second part is smart, based on the first answer:

- Same type of bike lane on both sides ⇒ prompts which type ⇒ shows current interface with both sides filled with that type
- Different on each side ⇒ shows current interface, with prompt for each side
- No bike lane ⇒ shows current interface with both sides filled with "none"

Also, I think the one-way quest could benefit from an interface like this.

If any of these sound interesting, I'll open new issues for them.

**Comment by mnalis** [comment]
Interesting idea in principle (and it might work for advanced users), but I am not convinced such pictures would easily convey meaning (especially for new users who haven't had solved that quest hundreds of times already). Current implementation is also not perfect - initially it was a little confusing to me, but question marks in picture and starting animation (showing where to click) helped guide me, and after first click it was very clear what it is asking.


IMHO just displaying it visually like in this mockup is **not very clear what user should select** (even in current quest when solved, it is not perfectly clear on first sight that gray-brick-looking one is supposed to be sidewalk, especially as sometimes we have sett roads too over here!), and adding descriptive text in your example would be problematic (adding "sidewalk only on right side"  text below picture would have to change depending on orientation, and even then would work badly when image is rotated by random number of degrees). 


But perhaps it could be somehow improved with longer description explaining to the user what is being asked exactly and what each picture depicts (although I am not sure - I myself was not reading the description on several occasions, and I'm sure I'm not the only one).

And/or perhaps the pictures could be improved; adding many small (like question marks in current quest) [no pedestrian signs](https://www.keysigns.co.uk/images/no-pedestrians-signs-p1368-45370_zoom.jpg) on pedestrian side, and [pedestrian signs](https://thumbs.dreamstime.com/z/pedestrian-zone-sign-romanian-64324072.jpg) on sidewalk side might help. Or they might be to small to be intelligible, don't really know.. Does anybody have ideas how to better convey to user what s/he should select?

**Reply by smichel17** [reply]
> even in current quest when solved, it is not perfectly clear on first sight that gray-brick-looking one is supposed to be sidewalk

I agree. For me it is mostly not about the particular texture, but that I find the markings to be very subtle — I have to look closely to distinguish a side with a sidewalk from a side with no sidewalk. I think the meaning would be clear to me even if the surfaces looked nothing like reality, as long as they are visually distinct. What do you think?

![image](https://user-images.githubusercontent.com/6563664/112768625-62692f80-900c-11eb-8e80-5094f89bd8d7.png)

Another option is to label them, "Both sides / One side (see image) / One side (see image) / Neither side". That way both/none gives the information needed to recognize which one is the sidewalk side in the middle images.

**Reply by smichel17** [reply]
Or you could put a big :heavy_check_mark: / :x: on each side, like in the current dialog. I think that would help even if the sidewalk is hard to see.

![image](https://user-images.githubusercontent.com/6563664/112768858-955ff300-900d-11eb-9262-73396dab9259.png)

**Reply by mnalis** [reply]
soooo, problem with GitHub discussion is that things get lost forever :cry: 
I liked that checkbox idea on sidewalks to make it more visible (and understandable especially to newcomers)

**Reply by matkoniecz** [reply]
> soooo, problem with GitHub discussion is that things get lost forever cry

If someone see discussion resulting in some good and actionable idea - open an issue. I recently went through my discussions as I almost forget about some - where hours of effort went into discussions already.

**Reply by smichel17** [reply]
+1 — we should consider discussions something more like a chat room (but more structured & linkable), where it's more about talking through ideas until they are in a more concrete form; any good+actionable outcome should be captured in a new issue (or at least bookmarked locally, if you don't have the time to open a new issue right away).
